src/pages/details/DetailsPage.tsx

```diff
@@ -0,0 +1,71 @@
+import { useCallback, useEffect, useRef, useState } from 'react';
+import styles from './DetailsPage.module.css';
+import { Product } from '../../models';
+import { fetchDetails, URL } from '../../helpers/api';
+
+interface DetailsProps {
+  id: number | null;
+  onClose: () => void;
+}
+
+function DetailsPage({ id, onClose }: DetailsProps): JSX.Element {
+  const [product, setProduct] = useState<Product | null>(null);
+  const detailsRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    const fetchProductData = async () => {
+      try {
+        const response = await fetchDetails(`${URL.base}/${id}`);
+        setProduct(response);
+      } catch (error) {
+        console.error('Error fetching product data:', error);
+      }
+    };
+    if (id) {
+      fetchProductData();
```

Comment:
Promise returned from fetchProductData is ignored :
```suggestion
      fetchProductData().then(() => {});
```